relay/nakama/handlers.go

```diff
@@ -193,15 +264,23 @@ func handleGetSaveGame(
 	nk runtime.NakamaModule,
 	_ string,
 ) (string, error) {
+	ctx, span := otel.Tracer("nakama.rpc").Start(ctx, "nakama/get-save")
+	defer span.End()
+
+	span.AddEvent("Reading save data")
 	result, err := readSave(ctx, nk)
 	if err == nil {
+		span.SetStatus(otelcode.Ok, "successfully retrieved saved game")
 		return utils.MarshalResult(logger, result)
 	}
 
+	span.RecordError(err)
 	if errors.Is(err, ErrNoSaveFound) {
+		span.SetStatus(otelcode.Error, "Failed to read save data")
 		return utils.LogErrorWithMessageAndCode(logger, err, codes.FailedPrecondition, "failed to read save data")
 	}
 
+	span.SetStatus(otelcode.Error, "Unknown error")
 	return utils.LogError(logger, err, codes.FailedPrecondition)
 }
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines +260 to +285]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->i tried to enable the linter 
- record-error
- set-status
and got this error 
`return can be reached without calling span.RecordError` and `span.RecordError is not called on all paths`

My suggestion is to always return the happy path with nil error.

for the example :
```
func handleGetSaveGame(
	ctx context.Context,
	logger runtime.Logger,
	_ *sql.DB,
	nk runtime.NakamaModule,
	_ string,
) (string, error) {
	ctx, span := otel.Tracer("nakama.rpc").Start(ctx, "nakama/get-save")
	defer span.End()

	span.AddEvent("Reading save data")
	result, err := readSave(ctx, nk)

	// Handle the case where there's an error
	if err != nil {
		// Always record the error first
		span.RecordError(err)

		// Handle the case where there's no save data found
		if errors.Is(err, ErrNoSaveFound) {
			span.SetStatus(otelcode.Error, "Failed to read save data")
			return utils.LogErrorWithMessageAndCode(logger, err, codes.FailedPrecondition, "failed to read save data")
		}

		// Unknown error case
		span.SetStatus(otelcode.Error, "Unknown error")
		return utils.LogError(logger, err, codes.FailedPrecondition)
	}

	// Marshal the result
	marshalResult, err := utils.MarshalResult(logger, result)

	// Handle the case where there's an error marshalling the result
	if err != nil {
		// Always record the error first
		span.RecordError(err)
		span.SetStatus(otelcode.Error, "Failed to marshal result")
		return utils.LogErrorWithMessageAndCode(logger, err, codes.FailedPrecondition, "failed to marshal result")
	}

	// If no error, mark span as OK and return the result
	span.SetStatus(otelcode.Ok, "successfully retrieved saved game")
	return marshalResult, nil
}
```

this will fix the linter error
<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-engine/799?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
thanks for the suggestion!